nightscout_osx_menubar.py

```diff
@@ -192,21 +213,80 @@ def get_history_menu_items(entries):
             time_ago=time_ago(seconds_ago(e['date'])),
         )
         for i, e in enumerate(bgs)
-    ][1:HISTORY_LENGTH + 1]
+    ][start:HISTORY_LENGTH + 1]
+
+websocket_entries = []
+websocket_status = "Connecting"
+socketIoClient = None
+
+class NightscoutNamespace(BaseNamespace):
+    def on_dataUpdate(self, *args):
+        for arg in args:
+            if type(arg) == dict:
+                print arg.keys()
+                global websocket_entries
+                for entry in arg.get('sgvs', ()):
+                    entry['date'] = entry['mills']
+                    entry['sgv'] = entry['mgdl']
+                    websocket_entries.append(entry)
+                websocket_entries = sorted(websocket_entries, key=lambda k: k['date'], reverse=True)
+        update_data(None)
+
+    def on_error(self, *args):
+        global websocket_status, socketIoClient
+        print "error", repr(args)
+
+        websocket_status = "Error: %s" % repr(args)
+        socketIoClient = None
+
+    def on_connect(self, *args):
+        global websocket_status
+        print "connected", repr(args)
+        websocket_status = WEBSOCKET_STATUS_CONNECTED
+        socketIoClient.wait_for_callbacks(1)
+
+    def on_disconnect(self, *args):
+        global websocket_status, socketIoClient
+        print "disconnected", repr(args)
+        websocket_status = "Disconnected"
+        socketIoClient = None
+
+
+@rumps.timer(WS_UPDATE_FREQUENCY_SECONDS)
+def service_websocket(sender):
+    global socketIoClient, websocket_status
+    try:
+        if socketIoClient is None:
```

Comment:
Looks like this should be initialized only if `config.get_use_websocket()` is true